Wvr manager prepare land

tiagovignatti opened the pull request:
Hi back! This continues my changes in preparation for the WebXR shared buffer transport work coming next.

Just like my previous PR, there are no functional changes in this PR. I've tested this using Meta Quest 2 on WebXR Samples and other immersive experiences from Wolvic init page. PTAL.

zakharvoit commented:
nit: I think some files are not formatted with Chromium style (I know that we didn't follow that too strictly in the past). I would suggest running "clang-format -i <filename>" at least for the new files but ideally for all changed files unless it generates too much noise.

zakharvoit reviewed (approved):
Approved but added a nit

svillar commented on wolvic/browser/vr/wvr_api.cc:
```diff
@@ -0,0 +1,125 @@
+
+namespace wolvic {
+
+WvrApi::WvrApi() {
```
I don't like the name either, but let's forget about it at the moment.

svillar commented on wolvic/browser/vr/wvr_api.h:
```diff
@@ -0,0 +1,48 @@
+  WvrApi(const WvrApi&) = delete;
+  WvrApi& operator=(const WvrApi&) = delete;
+
+  ~WvrApi();
```
just do `= default` here and you can get rid of the destructor in the cc file

svillar commented on wolvic/browser/vr/wvr_api.h:
```diff
@@ -0,0 +1,48 @@
+  WvrApi();
+
+  WvrApi(const WvrApi&) = delete;
+  WvrApi& operator=(const WvrApi&) = delete;
```
isn't there any macro in chromium to delete those? Anyway not very important

tiagovignatti replied:
I don't think so. Grepping for `& operator=(const ` shows thousands of similar occurrences.

svillar commented on wolvic/browser/vr/wvr_manager.h:
```diff
@@ -5,34 +5,29 @@
+namespace device {
+class MailboxToSurfaceBridge;
+}
+
```
Unrelated looks like

tiagovignatti replied:
this is intentional. Please refer to forward-declare nested classes: 
https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++.md#forward-declarations-vs_includes
https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-dos-and-donts.md#minimize-code-in-headers

svillar replied:
I know you're forward declaring, but as far as I saw you are still not using it

tiagovignatti replied:
ah, I know what made the confusion: I actually have cleaned up some headers being included in this file without explicitly mentioning it. You will see a`- #include "components/webxr/mailbox_to_surface_bridge_impl.h"
` but GitHub doesn't actually show the `device::MailboxToSurfaceBridge` class being used bellow in `WvrManager` (it's used to define a pointer called `mailbox_bridge_`).

Sorry about not being clear about that.

tiagovignatti commented:
Unfortunately I couldn't find a better way to rebase this branch (`wvr-manager-prepare-land`) on top of the newest `wolvic_experimental`, so I've rebased locally and forced a push my changes into `wvr-manager-prepare-land`.

Regardless, I believe I've addressed all the review comments now. I'll proceed and merge. Thank you!
